src/test/java/com/truelayer/java/acceptance/PaymentsAcceptanceTests.java

```
@@ -396,7 +398,12 @@ public void shouldCreateAPaymentRefundAndGetRefundDetails() {
                 tlClient.submitPaymentReturnParameters(submitProviderReturn).get();
         assertNotError(submitPaymentReturnParametersResponse);
 
-        waitForPaymentToBeSettled(paymentId);
+        try {
+            waitForPaymentToBeSettled(paymentId);
+        } catch (ConditionTimeoutException exception) {
+            Logger.warn("Wating for payment to be settled timed out, skipping test as refund cannot be created");
+            return;
```

I noticed this exception was thrown a few times even though the mock provider returns `execution_delay_in_seconds=2&settlement_delay_in_seconds=3`. To avoid introducing flaky tests this logic has been added to skip the test if the payment does not settle in time (if payment is not settled, refund call will fail)

---

As per a discussion with @dili91 we decided to don't swallow the exception, so let the test to fail but add a log stating the failure reason